horovod/spark/torch/remote.py

```diff
@@ -37,8 +37,8 @@
 
 def RemoteTrainer(estimator, metadata, last_checkpoint_state, run_id, dataset_idx):
     # Estimator parameters
-    loss_fns_pre_train = estimator.getLoss()
-    loss_constructors = estimator.getLossConstructors()
+    loss_fns_pre_train = to_list(estimator.getLoss())
```

Comment:
In https://github.com/horovod/horovod/blob/master/horovod/spark/torch/remote.py#L522 we pad the list of loss fns on every call to `calculate_loss`.  Why not do it here?  We could use `num_labels` to ensure the loss / loss_constructors / loss_weights are the right length.  We could even add it to `to_list`:

```
loss_fns_pre_train = to_list(estimator.getLoss(), length=num_labels)
```

Reply:
Moved it.